odl/solvers/advanced/chambolle_pock.py

```diff
@@ -0,0 +1,278 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""First-order primal-dual algorithm developed by Chambolle and Pock."""
```

Comment:
This style is perfectly OK:

```
"""First-order primal-dual algorithm developed by Chambolle and Pock.

The method was introducted in [REFERENCE] and is mostly used to solve [WHAT?], 
it has good convergence properties in case of [SOMETHING].
"""
```

As usual the first line should be concise, but after that we can have some more. No need for an essay though.